publishing-sdk/src/main/java/com/ably/tracking/publisher/workerqueue/results/WorkResult.kt

```diff
@@ -0,0 +1,49 @@
+package com.ably.tracking.publisher.workerqueue.results
+
+import com.ably.tracking.TrackableState
+import com.ably.tracking.common.ResultHandler
+import com.ably.tracking.publisher.Trackable
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+
+typealias AsyncWork<T> = (suspend () -> T)
```

Comment:
Two things:
1. This should be an `internal` type alias :wink: 
2. I think we can drop the generic type `T` from this and simply change it to an alias for a suspending function that returns a work result :wink: 
```suggestion
internal typealias AsyncWork = (suspend () -> WorkResult)
```

Reply:
AsyncWork was initally designed to reference any suspending work and not referring to any of our works. However I think it's a good idea to narrow it to WorkResult as we try to narrow the type scopes in this API
https://github.com/ably/ably-asset-tracking-android/pull/525/commits/d379881e191630b3e691237e18c34978dbce6869 and https://github.com/ably/ably-asset-tracking-android/pull/525/commits/b285571e06adf538dcc2ecc5c32409926d51d00e